resources/views/external-link-confirm.blade.php

```diff
@@ -1,66 +1,119 @@
+                            <x-ark-icon
+                                name="warning"
+                                size="xs"
+                            />
 
-        @slot('description')
-            <div class="flex flex-col mt-8 space-y-4 whitespace-normal">
-                <div class="font-semibold text-theme-secondary-900">
-                    <x-ark-alert type="warning" :message="$url" message-class="break-all" />
+                        </div>
+                    </div>
+                    <div class="alert-content-wrapper alert-warning-content">
+                        <span class="block leading-6 break-all" x-text="url"></span>
+                    </div>
                 </div>
-                <p>@lang('generic.external_link_disclaimer')</p>
             </div>
-        @endslot
 
-        @slot('buttons')
-            <button
-                class="button-secondary"
-                @click="hide"
-            >
-                @lang('actions.back')
-            </button>
+            <p>@lang('generic.external_link_disclaimer')</p>
 
-            <a
-                target="_blank"
-                rel="noopener nofollow"
-                class="cursor-pointer button-primary"
-                href="{{ $url }}"
-                @click="hide"
+            <x-ark-checkbox
+                name="confirmation"
+                alpine="toggle"
             >
-                @lang('actions.follow_link')
-            </a>
-        @endslot
-    </x-ark-js-modal>
-@endpush
+                @slot('label')
+                    @lang('ui::forms.do_not_show_message_again')
+                @endslot
+            </x-ark-checkbox>
+        </div>
+    @endslot
+
+    @slot('buttons')
+        <button
+            class="button-secondary"
+            @click="hide"
+        >
+            @lang('actions.back')
+        </button>
+
+        <a
+            target="_blank"
+            rel="noopener nofollow"
+            class="cursor-pointer button-primary"
+            :href="url"
+            @click="followLink()"
+        >
+            @lang('actions.follow_link')
+        </a>
+    @endslot
+</x-ark-js-modal>
+
+
+<script type="text/javascript">
+const initExternalLinkConfirm = () => {
+    const selectors = [
+        '[href*="://"]',
+        ':not([href^="{{ config("app.url") }}"])',
+        ':not([data-safe-external])',
+        ':not([data-external-link-confirm])',
+    ];
+
+    const links = document.querySelectorAll(`a${selectors.join('')}`);
+    links.forEach(link => {
+        link.setAttribute('data-external-link-confirm', 'true');
+        link.addEventListener('click', e => {
+            if (localStorage.getItem('has_disabled_link_warning') === 'true') {
+                return;
+            }
+
+            e.preventDefault();
+
+            Livewire.emit('openModal', 'external-link-confirm', link.getAttribute('href'));
+        });
+    });
+}
```

Comment:
@alfonsobries the issue is here. When minimized this results in `}document.addEventListener(...` which results in the JS not running properly